sbt_common/elasticsearch/src/test/scala/uk/ac/wellcome/elasticsearch/WorksIndexTest.scala

```diff
@@ -1,68 +1,68 @@
 package uk.ac.wellcome.elasticsearch
 
 import com.sksamuel.elastic4s.http.ElasticDsl.{indexInto, search}
-import org.scalacheck.Arbitrary
+import org.scalacheck.{Arbitrary, Shrink}
 import org.scalatest.concurrent.{Eventually, ScalaFutures}
 import org.scalatest.{BeforeAndAfterEach, FunSpec, Matchers}
 import uk.ac.wellcome.models._
 import uk.ac.wellcome.elasticsearch.test.fixtures.ElasticsearchFixtures
-import uk.ac.wellcome.test.utils.JsonTestUtil
+import uk.ac.wellcome.test.utils.{ExtendedPatience, JsonTestUtil}
 import uk.ac.wellcome.utils.GlobalExecutionContext.context
 import org.scalacheck.ScalacheckShapeless._
 import com.sksamuel.elastic4s.http.ElasticDsl._
 import org.elasticsearch.client.ResponseException
 import org.scalatest.prop.PropertyChecks
 import uk.ac.wellcome.utils.JsonUtil._
-import uk.ac.wellcome.utils.JsonUtil
-import uk.ac.wellcome.test.utils.ExtendedPatience
 
 class WorksIndexTest
     extends FunSpec
     with ElasticsearchFixtures
     with ScalaFutures
     with Eventually
-    with ExtendedPatience
     with Matchers
     with JsonTestUtil
     with BeforeAndAfterEach
+    with ExtendedPatience
     with PropertyChecks {
 
-  implicitly[Arbitrary[IdentifiedWork]]
+  // On failure, scalacheck tries to shrink to the smallest input that causes a failure.
+  // With IdentifiedWork, that means that it never actually completes.
```

Comment:
???

Feels to me there’s a fairly natural order of shrinking on case classes, so I’d be interested to understand why ScalaCheck is deficient here. I’ll do some reading.

👍 for comment